src/s3/commands/uploadFile.ts

```diff
@@ -26,51 +34,53 @@ export interface FileSizeBytes {
- * Uploads the file (showing a progress bar).
- * Shows the output channel with "upload completed" message.
- * Refreshes the node.
- *
- * Node that the node is reset to displaying its first page of results.
- * The file that is uploaded won't necessary fall on the first page.
- * The user may need to load more pages to see the uploaded file reflected in the tree.
+ * Uploads given file to given bucket, folder path is already in the key
+ * Refreshes explorer to show file uploaded 
+ * TODOD:: add documentation
+ * @param fileLocation - file to upload
+ * @param key - key to be used to upload
+ * @param bucket - bucket to upload to
+ * @param s3Client - owner of the bucket
+ * 
  */
-export async function uploadFileCommand(
-    node: S3BucketNode | S3FolderNode,
+ export async function uploadFileCommand(
+    fileLocation: vscode.Uri,
+    key: string,
+    bucket: S3.Bucket,
+    s3Client: S3Client,
     fileSizeBytes: FileSizeBytes = statFile,
     window = Window.vscode(),
-    commands = Commands.vscode(),
-    outputChannel = ext.outputChannel
-): Promise<void> {
-    getLogger().debug('UploadFile called for %O', node)
-
-    const fileLocation = await promptForFileLocation(window)
-    if (!fileLocation) {
-        getLogger().info('UploadFile cancelled')
-        telemetry.recordS3UploadObject({ result: 'Cancelled' })
-        return
-    }
-
+    outputChannel = ext.outputChannel,
+): Promise<void>{
+    
     const fileName = path.basename(fileLocation.fsPath)
-    const key = node.path + fileName
-    const destinationPath = readablePath({ bucket: node.bucket, path: key })
+    const destinationPath = readablePath({ bucket: {name: bucket.Name!}, path: key })
+    
     try {
         showOutputMessage(`Uploading file from ${fileLocation} to ${destinationPath}`, outputChannel)
+        const request = {
+            bucketName: bucket.Name!,
+            key: key,
+            fileLocation: fileLocation,
+            fileSizeBytes: fileSizeBytes(fileLocation),
+            s3Client,
+            window: window
+        }
 
-        await uploadWithProgress({ node, key, fileLocation, fileSizeBytes: fileSizeBytes(fileLocation), window })
-
-        showOutputMessage(`Successfully uploaded file ${destinationPath}`, outputChannel)
+        await uploadWithProgress(request)
+        showOutputMessage(`Successfully uploaded file ${destinationPath} to ${bucket.Name!}`, outputChannel)
         telemetry.recordS3UploadObject({ result: 'Succeeded' })
+        recordAwsRefreshExplorer()
+        vscode.commands.executeCommand('aws.refreshAwsExplorer')
+        return
+
     } catch (e) {
         getLogger().error(`Failed to upload file from ${fileLocation} to ${destinationPath}: %O`, e)
         showErrorWithLogs(localize('AWS.s3.uploadFile.error.general', 'Failed to upload file {0}', fileName), window)
         telemetry.recordS3UploadObject({ result: 'Failed' })
+        return
     }
-
-    await refreshNode(node, commands)
+    
```

Comment:
why was refreshNode() removed?

Reply:
It is taken care by `vscode.commands.executeCommand('aws.refreshAwsExplorer')` inside the try block